src/collections/map.rs

```diff
@@ -1007,50 +1223,156 @@ impl<K, V> TreeMap<K, V> {
     }
 }
 
-impl<K, V> Drop for TreeMap<K, V> {
+impl<K: AsBytes, V, const NUM_PREFIX_BYTES: usize, H: NodeHeader<NUM_PREFIX_BYTES>>
+    RawTreeMap<K, V, NUM_PREFIX_BYTES, H>
+{
+    /// Tries to get the given key’s corresponding entry in the map for in-place
+    /// manipulation.
+    pub fn try_entry(
+        &mut self,
+        key: K,
+    ) -> Result<Entry<K, V, NUM_PREFIX_BYTES, H>, InsertPrefixError>
+    where
+        K: AsBytes,
+    {
+        let entry = match self.root {
+            Some(root) => {
+                let insert_point = unsafe { search_for_insert_point(root, &key)? };
```

Comment:
typically for `unsafe` I always leave a comment `// SAFETY: ` which explains what the invariants are and why they are satisfied. I should probably go through and add a commit which does this for the newly added uses of `unsafe`

Reply:
A few months ago I added comments to good portion of the `unsafe`s but I might have missed this one

Reply:
But yeah, I would be good to double check

Reply:
It might be a different branch, since I'm seeing a lot of uncommented unsafe in this one

Reply:
I usually write stuff and after I go back fixing/adding comments, so this ones might be newer ones that I added and did't comment over it